src/webgpu/api/validation/createRenderPipeline.spec.ts

```diff
@@ -248,6 +249,56 @@ g.test('color_formats_must_be_renderable')
     t.doCreateRenderPipelineTest(isAsync, info.renderable && info.color, descriptor);
   });
 
+g.test('formats_for_depth_stencil_state_must_be_depth_stencil_formats')
+  .params(u => u.combine('isAsync', [false, true]).combine('format', kTextureFormats))
+  .fn(async t => {
+    const { isAsync, format } = t.params;
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    const descriptor = t.getDescriptor({ depthStencil: { format } });
+
+    t.doCreateRenderPipelineTest(isAsync, info.depth || info.stencil, descriptor);
+  });
+
+g.test('format_aspect_requirement_for_depth_stencil_test')
+  .desc(
+    `Tests the format aspect requirement:
+  - Depth aspect must be contained in the format if depth test or depth write is enabled.
+  - Stencil aspect must be contained in the format if stencil test or stencil write is enabled.
+  `
+  )
+  .params(u => u.combine('isAsync', [false, true]).combine('format', kDepthStencilFormats))
+  .fn(async t => {
+    const { isAsync, format } = t.params;
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    // Depth test is enabled
+    const descriptorDepthTest = t.getDescriptor({
```

Comment:
Can we test for all depth comparison fuctions? ("always" should be allowed IIRC?)

Ditto below for stencil comparison and fail op where some of the values are still allowed because they do nothing (I suggest splitting the test in two for depth and stencil)

Reply:
> "always" should be allowed IIRC?

Yes, according to the spec. (This is unlike blending which can be turned off separately. For depth/stencil test we turn them off implicitly iff everything is set to "always"/"keep".)

It's not super critical to test "always" because it's actually the default value in IDL so it's hard to get wrong. But I think since it's easy to test all cases we should go ahead and do it. (The 8 GPUCompareFunction values and `undefined`.)

Reply:
Done. And I added all enums of CompareFunctions and StencilOperations into capability_info.ts